pkg/agent/client/client_test.go

```diff
@@ -344,6 +352,90 @@ func TestNewX509SVIDs(t *testing.T) {
 	assertConnectionIsNotNil(t, client)
 }
 
+func TestNewX509SVIDsFails(t *testing.T) {
```

Comment:
Most of the code in this test is the same as in `TestNewX509SVIDs`. Can we re-use the existing setup logic in `TestNewX509SVIDs` and refactor it to be a table-driven test with this new test case?

Reply:
Done at [5818be9](https://github.com/spiffe/spire/pull/4262/commits/5818be93f6e17adccd2a16d2a8a3faa4430479cd)